src/it/scala/uk/gov/hmrc/support/ScalaCheckUtils.scala

```diff
@@ -3,25 +3,29 @@ package uk.gov.hmrc.support
 import org.scalacheck.Prop
 import org.scalacheck.Prop.collect
 
+import scala.collection.compat.immutable.LazyList
+import scala.collection.compat.immutable.LazyList.#::
+
 object ScalaCheckUtils {
 
   type ClassifyParams = (Boolean, Any, Any)
 
   /**
     * Collect statistics after checking a property. See [[Prop.classify]]
-    * Convenience method that avoids nesting several [[Prop.classify]] by providing a [[Stream]] of conditions.
-    * A [[Stream]] is used to avoid evaluating the conditions prematurely.
+    * Convenience method that avoids nesting several [[Prop.classify]] by providing a [[LazyList]] of conditions.
+    * A [[LazyList]] is used to avoid evaluating the conditions prematurely.
     *
-    * @param conditions [[Stream]] of triples (condition, ifTrue, ifFalse)
+    * @param conditions [[LazyList]] of triples (condition, ifTrue, ifFalse)
     * @param prop       a scalacheck property [[Prop]]
     * @return [[Prop]]
     */
   @scala.annotation.tailrec
-  def classify(conditions: Stream[ClassifyParams])(prop: Prop): Prop = conditions match {
-    case Stream.Empty       => prop
-    case h #:: Stream.Empty =>
+  def classify(conditions: LazyList[ClassifyParams])(prop: Prop): Prop = conditions match {
+    case LazyList()       => prop
+    case h #:: LazyList() =>
       if (h._1) collect(h._2)(prop) else collect(h._3)(prop)
-    case h #:: t            =>
+    case h #:: t          =>
       classify(t)(Prop.classify(h._1, h._2, h._3)(prop))
+    case _                => throw new MatchError(s"Unexpected match error on $conditions")
```

Comment:
added this to suppress a phantom MatchError due to [a compiler bug](https://users.scala-lang.org/t/pattern-matching-lazylist/5105/10)